### PR TITLE
Improved ergonomy for `CREATE EXTERNAL TABLE OPTIONS`: Don't require quotations for simple namespaced keys like `foo.bar`

### PR DESCRIPTION
**Pull request by ozankabak**
## Which issue does this PR close?

Quick follow-on to #10404.

## Rationale for this change

Now that we migrated to a consistent options syntax for external tables, we should focus on user ergonomy. I don't see a reason why we require simple namespaced option keys like `foo.bar` to have explicit quotes.

## What changes are included in this PR?

Tweaked parser logic to allow quoteless namespaced keys for the `OPTIONS` clause.

## Are these changes tested?

Added a new slt test to exercise the new code path.

## Are there any user-facing changes?

Increased user ergonomy.

### CODE REVIEW

File: datafusion/sql/src/parser.rs [hunk 1]
```diff
@@ -462,7 +462,18 @@ impl<'a> DFParser<'a> {
...
+                while self.parser.consume_token(&Token::Period) {
+                    let next_token = self.parser.next_token();
+                    if let Token::Word(Word { value, .. }) = next_token.token {
+                        parts.push(value);
+                    } else {
+                        return self.parser.expected("key name", next_token);
+                    }
+                }
+                Ok(parts.join("."))
+            }
```
Comment (ozankabak) [inline datafusion/sql/src/parser.rs:476]: This is the main change, the rest is clean-up.

**Comment by ozankabak** [comment]
@berkaysynnada PTAL

File: datafusion/sqllogictest/test_files/create_external_table.slt [hunk 2]
```diff
@@ -201,7 +201,20 @@ CREATE EXTERNAL TABLE IF NOT EXISTS region (
...
+CREATE EXTERNAL TABLE IF NOT EXISTS region (
+        r_regionkey BIGINT,
+        r_name VARCHAR,
+        r_comment VARCHAR,
+        r_rev VARCHAR,
+) STORED AS CSV LOCATION 'test_files/tpch/data/region.tbl'
+OPTIONS (
+        format.delimiter '|',
+        has_header false,
+        compression gzip);
```
Comment (berkaysynnada) [inline datafusion/sqllogictest/test_files/create_external_table.slt:220]: 👍🏻

**Comment by berkaysynnada** [comment]
This PR addresses the last remaining gap, ensuring that UX achieves the desired balance of consistency and flexibility.

File: datafusion/sql/src/parser.rs [hunk 3]
```diff
@@ -462,7 +462,18 @@ impl<'a> DFParser<'a> {
...
         match next_token.token {
-            Token::Word(Word { value, .. }) => Ok(value),
+            Token::Word(Word { value, .. }) => {
+                let mut parts = vec![value];
+                while self.parser.consume_token(&Token::Period) {
+                    let next_token = self.parser.next_token();
+                    if let Token::Word(Word { value, .. }) = next_token.token {
+                        parts.push(value);
+                    } else {
+                        return self.parser.expected("key name", next_token);
```
[the remark this answers is not in the document]
Comment (ozankabak) [reply datafusion/sql/src/parser.rs:475]: Sure, thanks for taking a look 👍

File: datafusion/sqllogictest/test_files/create_external_table.slt (hunk 2 above)
Comment (alamb) [reply datafusion/sqllogictest/test_files/create_external_table.slt:220]: ❤️

**Review by alamb — COMMENTED** [review]
Thanks @ozankabak  and @berkaysynnada  and @comphead  -- this looks really nice 👍

**Comment by ozankabak** [comment]
Thank you all for the reviews 🚀

**Pull request merged**